Specalc.package/Specalc.class/instance/defineDigitButtons.st

```diff
@@ -1,14 +1,7 @@
 button-defintion
 defineDigitButtons
-	button0
-		label: '0';
-		action: [ display label: ((display label,'0') asNumber) asString];
-		state: true.
-		
-	button1
-		label: '1';
-		action:  [ display label: ((display label,'1') asNumber) asString ];
-		state: true.
+	button0 beButton: 0 for: self.	
```

Comment:
So you can see how this method is starting to shrink and become more intention revealing. At the extreme end, one could use metaprogramming to *really* consolidate, but the savings is usually outweighed by added opaqueness, and the technique reserved mostly for tooling. But it would look like:
```
0 to: 9 do: [ :e | | var accessor |
	accessor := ('button,' e asString) asSymbol.
	var := self perform: accessor.
	var beButton: e for: self ].
```